primitives/Cargo.toml

```diff
@@ -4,22 +4,29 @@ version = "0.6.0"
 authors = ["Supercomputing Systems AG <info@scs.ch>"]
 license = "Apache-2.0"
 edition = "2021"
+repository = "https://github.com/scs/substrate-api-client"
+description = "Substrate-api-client primitive types"
+readme = "README.md"
+categories = ["no-std"]
+
 
 [dependencies]
 codec = { package = "parity-scale-codec", version = "3.0.0", default-features = false, features = ['derive'] }
-hex = { version = "0.4.3", default-features = false, features = ["alloc"] }
 impl-serde = { version = "0.4", default-features = false }
 primitive-types = { version = "0.12.1", default-features = false, features = ["serde_no_std", "scale-info"] }
 scale-info = { version = "2.1.1", default-features = false, features = ["derive"] }
 serde = { version = "1.0", default-features = false, features = ["derive", "alloc"] }
 serde_json = { version = "1.0", default-features = false, features = ["alloc"] }
 
 # substrate no_std
-sp-core = { default-features = false, git = "https://github.com/paritytech/substrate.git", branch = "master" }
+sp-core = { default-features = false, features = ["full_crypto"], git = "https://github.com/paritytech/substrate.git", branch = "master" }
 sp-runtime = { default-features = false, git = "https://github.com/paritytech/substrate.git", branch = "master" }
+sp-runtime-interface = { default-features = false, git = "https://github.com/paritytech/substrate.git", branch = "master" }
 sp-staking = { default-features = false, git = "https://github.com/paritytech/substrate.git", branch = "master" }
 sp-version = { default-features = false, git = "https://github.com/paritytech/substrate.git", branch = "master" }
 sp-weights = { default-features = false, git = "https://github.com/paritytech/substrate.git", branch = "master" }
+# need to add this for the app_crypto macro
```

Comment:
This is not quite clear to me. Can you provide a bit more context?

Reply:
The problem is that `sp-application-crypto` is indirectly imported by some other crate of substrate, but without the feature `full_crypto`. To get it working, however, we need this feature, so we import it directly to activate it.

I haven't looked too deeply into what this app_crypto thing is, but there is a `AppCrypto` trait here: https://github.com/paritytech/substrate/blob/master/primitives/application-crypto/src/traits.rs#L34
And it is a fact that it does not compile without activating the full_crypto feature.